### PR TITLE
Adds new T2 Xeno, new toxins, adds damagetype to human_damage_taken signal

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request

Afflictor - stealthy T2, evolves from Sentinel into Defiler

The afflictor is a xeno that trades direct combat with applying debuffs in the form of toxins that weaken and disrupt the marines so that other xenos can have better engagements. With this, I am aiming to make a caste where the aspect is put on cooperation and working around other xenos rather than having straight combat capability.

Chemicals - Applied in doses of 15u and last 1m without purging reagents in effect
 - Hemodile - applies 25% slow (increased to 50% if Praelyx or Neurotoxin are present) after ~5s increases stamina damage taken by 50%
 - Transvitox - after ~5s converts 50% brute/burn damage received to 60% toxin damage up to 45 (puke threshold)
 - Praelyx - unlocked at amcient - injected once, makes reagent slash deal 10 brute to targeted limb (not negated by armor) in exchange for curing 6 toxin

Abilities:
Select Reagent- selects a toxin similarly to resin structures
Reagent Slash - 6s CD, 40 plasma - slashes 4 times with 1s delay between slashes for 10 brute damage and injects 4u of selected reagent per slash into a marine next to it. Both the xeno and marine can move during this time. Has aim assist as qol, so only the tile the marine is on needs to be clicked
Camouflage - 7s CD, 70 plasma initial cost, 1p while walking, 6p while running - When moving stealth is the same as hunter's. Becomes almost completely invisible when standing still - better than hunter, same as tarp. 
Reagent Sight -  able to see the reagents inside a marine and how much of each there is when inspecting them
Is also able to walk over half-walls/tables like runner

Stats:
Everything is the same as panther’s except the plasma values below and speed is equal to drone's:
        plasma_max = 100>180>220>260
	plasma_gain = 4>6>8>10

Ability icons are not yet in

## Why It's Good For The Game

I think this will be good for the game as it adds another way to play xenos as a type of an offensive support role. While most xenos fight by directly facing the enemy, this one will benefit from teaming up and helping guide the flow of engages. I hope this concept would also allow for more dynamic rounds since it encourages xenos to fight away from LZ and thus shifting around the map more.

## Changelog
:cl:Slotbot
add: Afflictor xeno files
add: new toxins
add: damagetype to human_damage_taken signal
add: xeno hud elements and examine text based on the new toxins in a marine
imageadd: Afflictor sprites, damage states and hud icons
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
oh god oh fuck it's neuro claws all over again

**Comment by neohenry** [comment]
In all seriousness, how do the marines counter the toxins? Would they need to use hypervene?

Jroin, we need you.

**Comment by Walarks** [comment]
> In all seriousness, how will the marines counter the toxins? Would they need to use hypervene?
> 
> Jroin, we need you.

Hemodile is only removed by purging it or waiting it out, though it won't do any damage if you don't get attacked. The others can be healed by their respective chems or purged by hypervene. Only decay accelerant deals damage over time over a longer duration.

File: code/modules/reagents/reactions/toxin.dm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+////// Xeno Chem reactions
+/datum/chemical_reaction/xeno_decaytoxin
+    results = list(/datum/reagent/toxin/xeno_decaytoxin = 4)
+    required_reagents = list(/datum/reagent/toxin/xeno_growthtoxin = 4, /datum/reagent/toxin/xeno_decaytoxin_catalyst = 0.1)
+    required_catalysts = list(/datum/reagent/toxin/xeno_decaytoxin_catalyst = 1)
+    mob_react = TRUE
```
Comment (XSlayer300) [inline code/modules/reagents/reactions/toxin.dm:6]: No newline at the end of the file.

File: code/modules/reagents/reagents/toxin.dm [hunk 2]
```diff
@@ -506,3 +507,139 @@
...
+	reagent_state = LIQUID
+	color = "#802400" // rgb: 128, 36, 0
+	custom_metabolism = REAGENTS_METABOLISM
+	overdose_threshold = REAGENTS_OVERDOSE
+	scannable = TRUE
+	toxpwr = 0
+
+/datum/reagent/toxin/xeno_decaytoxin/on_mob_life(mob/living/L, metabolism)
+	if(prob(6))
+		L.visible_message("<span class='warning'>We can feel our body falling apart!</span>")
```
Comment (XSlayer300) [inline code/modules/reagents/reagents/toxin.dm:596]:
```suggestion
		L.visible_message("<span class='warning'>You can feel your body falling apart!</span>")
```

**Comment by Chaznoodles** [comment]
All of these chems are countered by swallowing a single pill - Hypervene.

Proto Larval Decay is worthless because if someone is larval accelerant-ed, then they're probably infected and you shouldn't be killing them, plus drones aren't frontline, plus _why would a drone be healstinging people on the frontline_

Praelyx: Enjoy finding the same marine twice and stinging them for an almighty 16 brute cured by a single bruisepack or pill - if they didn't purge themself with hypervene.

Hemodile would be worthwhile, but haha hypervene!

Transvitox is the only worthwhile one here - but still isn't worth it because instead of using double the meds, they now only need to pop a Dylo pill.

All in all, this isn't worth a T2 slot.

File: code/modules/reagents/reactions/toxin.dm (hunk 1 above)
Comment (Walarks) [reply code/modules/reagents/reactions/toxin.dm]: should be ok now I think

**Comment by jroinc** [comment]
Thoughts on this (the chems only, cause that's what I'm qualified to discuss).

Hemodile- Takes too long to take effect for what it causes. 25% or 50% additional stamina damage (does armor affect the bonus damage?) isn't as meaningful without a clear indicator to other ayys to target DAT GUY (and disarms only, ples).

Transvistitox is an interesting idea that I like, but should probably begin happening instantly, as converting 2 damage types into 1 is meh. Sure, ungas don't normally carry antitox, but there's nothing really STOPPING them from doing so if the meta shifts (which this just made happen).

Praelyx is kinda a shitpost, remove it. "OOH, I have an ability that causes as much damage as ONE SLASH if I hit the marine with it twice within 20 seconds!"

Proto decay accelerant can be removed and just replaced with a direct injection of Decay accelerant. 1 brute and 1 tox/tick is an acceptable T2 ability, given as marines can and will counter your shit 100% of the time once they catch on.

Why do they have an overdose threshold if they don't do anything on OD, also?

Your metabolism rates and amounts are utterly unacceptable (and also not pinned to REAGENTS_METABOLISM, but I don't really care about that. Here's why.

In THEORY, 10u injected with a metabolism rate of 0.048 will last for 7 minutes or whatever it comes out to.
In PRACTICE, the unga will take 1 antitox pill. Which purges all toxins at 1u/tick. So it lasts for 18 seconds, and doesn't trigger.
And he's IMMUNE for the duration of the pill, which is a minute or two, and the bottle has 15 more in them.

To fix this (my ideas)-

Remove Praelex. I can't save it in any way.

Replace the "at young" ability with regular old neurotoxin (and add a 0.5/tick toxin damage). Simple and not flashy, but it gets the job done, it's toxin-y, AND it purges antitoxin, so you have some abilities against druggie ungas.

Move Transvistitox to Mature, keep the same effects, and give it the 1u/tick purge that neurotoxin has (as well).

Move Hemodile to Elder. Keep the current effects. In addition, add a 1u/tick bloodloss effect (I can show you the code if needed) so it does something even if people don't target the current guy (which will happen often, in fact, almost every time). Move the effects to happen instantly. Don't give it the antitox purge effect.

Replace the Proto Decay Accelerant sting with Decay Accelerant. Good ancient ability, don't give it the purge effect.

I would recommend altering the reagent rates and amounts to fall into the same levels as neurotoxin, and give it the same channel ability as current stings (3 jabs of 10u each, with only the first one being instant).

**Comment by jroinc** [comment]
I was informed that the xeno cannot slash.

Trash this as a concept, it's garbage.

*Today I will make a xeno with an extremely fiddly way of hurting ungas that does NOTHING if they use any of a few extremely common medicines that they can easily access in unlimited quantities*

**Comment by TiviPlus** [comment]
Keep it civil or I send you into the stratosphere and see how well you can fly

**Comment by Walarks** [comment]
Seeing as there is more works that needs to be done and I got more ideas on changing the sprite/chems/abilities. Will convert to draft for now.

**Comment by jroinc** [comment]
> Keep it civil or I send you into the stratosphere and see how well you can fly

Here's what I like about it-

-Minimum of buttons, so it's easy to select a poison.
-It has pounce, so it can reliably mess with one marine that's advancing.
-Near-full invisibility is very handy to have. The "You feel a tiny prick!" message will give you away, but you'll at least be able to reliably sting then escape.
-Reagent sight is a must-have for this caste, as it lets you know who to target.
-Base stats are acceptable for a T2, provided it has it's invisibility up at all times when approaching marines.

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 3]
```diff
@@ -86,6 +86,9 @@
 	var/aura_strength = 0 //The strength of our aura. Zero means we can't emit one
 	var/aura_allowed = list("frenzy", "warding", "recovery") //"Evolving" removed for the time being
 
+	// *** Afflictor Abilities *** //
+	var/list/available_reagents_define = list() //reagents available for select reagent
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:91]: Why cant this just stay on the castedatum as it is right now?

File: code/game/data_huds.dm [hunk 4]
```diff
@@ -209,6 +209,19 @@
...
+		xeno_reagent.icon_state = "transvitox_icon"
+
+	if(reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile))
+		xeno_reagent.icon_state = "hemodile_icon"
+
+	if(reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile) && reagents.get_reagent_amount(/datum/reagent/toxin/xeno_transvitox))
+		xeno_reagent.icon_state = "hemodile_transvitox_icon"
+
+	if(!reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile) && !reagents.get_reagent_amount(/datum/reagent/toxin/xeno_transvitox))
+		xeno_reagent.icon_state = ""
```
Comment (TiviPlus) [inline code/game/data_huds.dm:224]: dupe code

File: code/modules/reagents/reagents/toxin.dm [hunk 5]
```diff
@@ -506,3 +507,95 @@
...
+	scannable = TRUE
+	toxpwr = 0
+
+/datum/reagent/toxin/xeno_hemodile/on_mob_add(mob/living/L, metabolism, affecting)
+	to_chat(L, "<span class='warning'>You have trouble catching your breath.</span>")
+	RegisterSignal(L, COMSIG_HUMAN_DAMAGE_TAKEN, .proc/hemodile_human_damage_taken)
+
+/datum/reagent/toxin/xeno_hemodile/on_mob_life(mob/living/L, metabolism)
+	return ..()
+
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/toxin.dm:528]:
```suggestion
```

File: code/modules/reagents/reagents/toxin.dm [hunk 6]
```diff
@@ -506,3 +507,95 @@
...
+	scannable = TRUE
+	toxpwr = 0
+
+/datum/reagent/toxin/xeno_praelyx/on_mob_add(mob/living/L, metabolism, affecting)
+	if(L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_transvitox) || L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_decaytoxin) || L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile))
+		RegisterSignal(L, COMSIG_HIVE_XENO_RECURRING_INJECTION, .proc/xeno_praelyx_trigger)
+
+/datum/reagent/toxin/xeno_praelyx/on_mob_life(mob/living/L, metabolism)
+	return ..()
+
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/toxin.dm:596]:
```suggestion
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
just mostly stylistical concerns left so this is ok for tm

**Comment by davidchan13** [comment]
Given how vomit puking completely stuns you this caste is way more powerful than you might consider, 2 uses of their chems and a marine can be completely out of a fight more so than a marine on the verge of pain crit.

**Comment by jroinc** [comment]
Note about vomiting-

Vomiting starts to occur at 45 toxin damage (which is within 1 sting of transvitox's conversion ability).

Vomiting hardstuns for 10 seconds, and cannot be resisted or shook out of.

**Comment by AtlasRules** [comment]
this is genuinely awful to fight, please reconsider just about everything as getting 90 toxin damage out of nowhere is not fun and never will be

**Comment by jroinc** [comment]
> this is genuinely awful to fight, please reconsider just about everything as getting 90 toxin damage out of nowhere is not fun and never will be

It's not "out of nowhere".

The worst this can do is 1 toxin damage/tick if you're already hurt, until it hits ancient.

At ancient, it can do ~3-5 toxin/tick if you're unlucky, 1-2 more normally.

**Comment by neohenry** [comment]
I think we need to make the text bigger if you get stung. Given how scary this is, we should indicate better of whether marine got stung.

**Comment by Walarks** [comment]
In regards to the vomiting, I would like to mention that currently the conversion happens at after a ~5s delay and at 4 brute and 4 burn per second (since they are converted at the same time it happens faster if you have both brute and burn instead of only one). Best case scenario it will take 11s to achieve 46 toxin after being damaged by a different xeno. If you have only brute it will take ~17s.

**Comment by neohenry** [comment]
@Walarks , is there an ability lock on Reagent Slash after you get out of cloak? Was wondering since I sense that is; if not, that's just be being unrobust.

**Comment by Walarks** [comment]
Can check again, but it should be always usable as long as you have plasma. You can use it while stealthed so the person has less incentive to move and cause you to miss.

**Comment by jroinc** [comment]
Alright, played a few rounds, and here are my impressions-

Invisibility is good, one of the best things about the caste.

ReagentHUD is my personal favorite, lots of good diagnostics.

It's very, very fragile till ancient, to the point where I don't want to expose myself to fire at all.

Plasma economy is tight till ancient, but this can be avoided with drone coop, so that's fair.

It's a one-trick pony. Maybe weak acid would give you something to do when not trying to be a living landmine?

Stinging people is very, very fincky, might be better transferring it to a normal slash (neuro-claws style) with lower amounts. Or maybe I'm just a bad clicker, or maybe I should work closer with people with stuns. I'd love to see a larger sting message for the ayy, though, can be hard to tell if you actually stung.

You can nest-sting people for the KEKs.

Didn't use either Hemodile (not my style) or Praelex (I can't bring myself to use a chem that does 4 slashes worth of damage (counting armor!) in exchange for possible toxin damage from another of my poisons, opportunity cost is too great).

Transvitox- Hot take. It's ACTUALLY BAD for ayys, due to the following-

-If the unga doesn't have antitox, pulling 1 burn and 1 brute for 1.1 toxin is good.

-If the unga HAS antitox (which will be guaranteed soon for the robust ones), you're trading 1 brute OR 1 burn for 0.1 toxin, basically giving them another healing chem that stacks with their current meds. If they have both brute and burn, you gain a little, but it's still a 1 point net heal that they didn't have before (and stacks with their 2 point heal from bicard and kelo).

Decay accel- THE reason to pick the class. I stung someone who was on 6 meds and watched them die within like 20 seconds. ~then they got recovered and revived, but oh well~

1 toxin and 1 brute on a non-medded person's nothing to sneeze at either, and someone on the big-3 meds is hitting 4 damage (4 toxin, 0 brute)/tick.

One issue- it kills caps. 15u processes for 60 cycles. That's 60 toxin and 60 brute. Assuming the target is larval stung, half of that's healed for a total of 60 damage, which will kill the *usual* capture. Maybe make larval toxin purge it?

My overall impression is that you pick it, go AFK/very, VERY carefully skirmish and do minor damage till ancient, then you pick Decay Accel and you're a hunter sidegrade.

**Comment by jroinc** [comment]
FYI-

Antitox purge rate was dropped from 2/tick to 1/tick in another PR, I think I already told you this.

**Comment by Walarks** [comment]
So, I changed the reagent slash to be 4 attacks, each applying a portion of the total reagent amount. This should allow marines a change to react and fight the xeno when it happens. Also it should work better with the antitox purge rate being reduced as it should allow lower doses to have some effect vs being completely negated.

Also will look into giving "aim assist" to reagent slash or infusing it into normal slashes next vs having to click on marine's sprite.

**Comment by jroinc** [comment]
> Also will look into giving "aim assist" to reagent slash or infusing it into normal slashes next vs having to click on marine's sprite.

https://github.com/tgstation/TerraGov-Marine-Corps/pull/3685/files may be of use, as that's the shit that enabled slashes applying reagents. Whether it should be a thing is another question.

File: code/modules/mob/living/carbon/xenomorph/castes/afflictor/abilities_afflictor.dm [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+/datum/action/xeno_action/xeno_camouflage
+	name = "Toggle Nanocrystal Camouflage"
+	action_icon_state = "stealth_on"
+	mechanics_text = "Become harder to see, better camouflage when walking and almost invisible if you stand still. Uses plasma to move, more when running."
+	ability_name = "stealth"
+	plasma_cost = 10
+	keybind_signal = COMSIG_XENOABILITY_TOGGLE_NANOCRYSTAL_CAMOUFLAGE
+	COOLDOWN_DECLARE(xeno_camouflage_cooldown)
+	var/can_sneak_attack = FALSE
+	var/last_stealth = null
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/afflictor/abilities_afflictor.dm:101]: Comment last_stealth to explain why it isnt a cooldown and set it to 0

File: code/modules/mob/living/carbon/xenomorph/castes/afflictor/afflictor.dm [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+
+
+// ***************************************
+// *********** Mob overrides
+// ***************************************
+
+/mob/living/carbon/xenomorph/afflictor/apply_alpha_channel(image/I)
+	I.alpha = src.alpha
+	return I
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/afflictor/afflictor.dm:26]:
```suggestion
	I.alpha = alpha
	return I
```

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 9]
```diff
@@ -590,30 +590,42 @@
...
-		to_chat(src, "<span class='xenowarning'>Our stinger injects our victim with [body_tox.name]!</span>")
+		if(is_reagent_slash)
+			if(get_dist(src, C) > 1)
+				to_chat(src, "<span class='warning'>We need to be closer to [C].</span>")
+				return
+			to_chat(src, "<span class='xenonotice'>We slash [C] with [body_tox.name]!</span>")
+			playsound(C, "alien_claw_flesh", 15, TRUE)
+			playsound(C, 'sound/effects/spray3.ogg', 15, TRUE)
+			C.apply_damage(damage = 4, damagetype = BRUTE, def_zone = affecting, sharp = TRUE)
+			to_chat(C, "<span class='danger'>The [src] swipes at you!</span>")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:618]: These two tochats should be a visible_message

File: code/modules/reagents/reagents/toxin.dm [hunk 10]
```diff
@@ -476,6 +476,7 @@
 /datum/reagent/toxin/xeno_neurotoxin/overdose_crit_process(mob/living/L, metabolism)
 	L.Losebreath(1) //Can't breathe; for punishing the bullies
 
+
 /datum/reagent/toxin/xeno_growthtoxin
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/toxin.dm:480]:
```suggestion
/datum/reagent/toxin/xeno_growthtoxin
```

File: code/modules/reagents/reagents/toxin.dm [hunk 11]
```diff
@@ -506,3 +507,90 @@
...
+	description = "A stamina draining toxin. Causes increasing stamina loss the longer it is present."
+	reagent_state = LIQUID
+	color = "#94CFD8" // rgb: 148, 207, 216
+	custom_metabolism = 0.5
+	overdose_threshold = 10000
+	scannable = TRUE
+	toxpwr = 0
+
+/datum/reagent/toxin/xeno_hemodile/on_mob_add(mob/living/L, metabolism, affecting)
+	to_chat(L, "<span class='warning'>You have trouble catching your breath.</span>")
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/toxin.dm:523]: On mob add?
This should probably be change to more "you suddenly feel your lungs contract" to indicate it was just injected or moved to a prob() in metabolize

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — DISMISSED** [review]

File: code/__DEFINES/dcs/signals.dm [hunk 12]
```diff
@@ -279,7 +279,7 @@
 
 // /mob/living/carbon/human signals
 #define COMSIG_HUMAN_MELEE_UNARMED_ATTACK "human_melee_unarmed_attack"	//from mob/living/carbon/human/UnarmedAttack(): (atom/target)
-#define COMSIG_HUMAN_DAMAGE_TAKEN "human_damage_taken"					//from human damage receiving procs: (mob/living/carbon/human/wearer, damage)
+#define COMSIG_HUMAN_DAMAGE_TAKEN "human_damage_taken"					//from human damage receiving procs: (mob/living/carbon/human/victim, damage, damagetype)
```
Comment (spookydonut) [inline code/__DEFINES/dcs/signals.dm:282]: you cant change signals like this.

File: code/modules/mob/living/carbon/human/species.dm [hunk 13]
```diff
@@ -982,7 +982,7 @@
 			victim.adjustStaminaLoss(damage)
 
 	// Will set our damageoverlay icon to the next level, which will then be set back to the normal level the next mob.Life().
-	SEND_SIGNAL(victim, COMSIG_HUMAN_DAMAGE_TAKEN, damage)
+	SEND_SIGNAL(victim, COMSIG_HUMAN_DAMAGE_TAKEN, damage, damagetype)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/human/species.dm:985]: cant do this

**Review by spookydonut — CHANGES_REQUESTED** [review]
theres 3 usages of this signal, you can't change the existing signal like this.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 14]
```diff
@@ -20,6 +20,26 @@
 			msg += "<span style='font-weight: bold; color: purple;'>You sense this creature is dead.</span>\n"
 		else if(stat || !client)
 			msg += "<span class='xenowarning'>It doesn't seem responsive.</span>\n"
+		if(!isxenoafflictor(user))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/human/examine.dm:23]: gross!

Comment (spookydonut) [reply code/modules/mob/living/carbon/human/examine.dm:23]: this is super snowflakey

File: code/modules/mob/living/carbon/xenomorph/castes/afflictor/abilities_afflictor.dm [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+		X.visible_message(C, "<span class='danger'>The [X] swipes at [C]!</span>")
+		X.do_attack_animation(C)
+	while(i++ < count && do_after(X, channel_time, TRUE, C, BUSY_ICON_HOSTILE, ignore_turf_checks = TRUE, target_can_move = TRUE))
+	return TRUE
+
+
+// ***************************************
+// *********** NANOCRYSTAL CAMOUFLAGE
+// ***************************************
+/datum/action/xeno_action/xeno_camouflage
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/castes/afflictor/abilities_afflictor.dm:121]: this looks like a copy paste of hunter stealth but not, why is this intruding on hunter niche?

File: code/modules/reagents/reagents/toxin.dm [hunk 16]
```diff
@@ -506,3 +506,70 @@
 
 /datum/reagent/toxin/xeno_growthtoxin/overdose_crit_process(mob/living/L, metabolism)
 	L.Losebreath(2)
+
+
+/datum/reagent/toxin/xeno_hemodile
```
Comment (spookydonut) [inline code/modules/reagents/reagents/toxin.dm:511]: no way to counter this as a medic/doctor

no adverse reactions with other chems either?

**Review by spookydonut — CHANGES_REQUESTED** [review]

**Comment by spookydonut** [comment]
While this has new keybinds, this cannot be testmerged.

**Comment by spookydonut** [comment]
I'm against adding any new castes.

most or all of these abilities could be added to reinforce existing caste niches

File: code/modules/reagents/reagents/toxin.dm (hunk 16 above)
Comment (Walarks) [reply code/modules/reagents/reagents/toxin.dm:511]: It is counterable through dylovene and hypervene that purge toxins.

**Comment by Walarks** [comment]
code bad

**Pull request closed without merging**